src/main/java/nl/tudelft/lifetiles/graph/view/TileView.java

```diff
@@ -66,14 +54,17 @@ public final void changeVertexColor(final VertexView v, final Color color) {
     /**
      * Draw the given graph.
      *
-     * @param gr
+     * @param segments
      *            Graph to be drawn
      * @return the elements that must be displayed on the screen
      */
-    public final Group drawGraph(final Graph<SequenceSegment> gr) {
-        PriorityQueue<SequenceSegment> it = sortStartVar(gr);
-        while (!it.isEmpty()) {
-            SequenceSegment segment = it.poll();
+    public final Group drawGraph(final SortedSet<SequenceSegment> segments) {
+        root = new Group();
+        nodes = new Group();
+        edges = new Group();
+        lanes = new LinkedList<Long>();
```

Comment:
Why a `LinkedList`? You use it mainly by index so an `ArrayList` would be faster.